app/extractors/BeneficiaryExtractor.scala

```diff
@@ -65,6 +84,21 @@ trait BeneficiaryExtractor[T <: Beneficiary] {
       .flatMap(answers => extractAddress(entity.address, answers))
   }
 
+  def extractUserAnswersForIndBeneficiary(answers: UserAnswers,
+                                          entity: IndBeneficiary): Try[UserAnswers] = {
+    answers
+      .set(fullNamePage, entity.name)
+      .flatMap(_.set(roleInCompanyPage, entity.roleInCompany))
+      .flatMap(_.set(mentalCapacityYesNoPage, entity.mentalCapacityYesNo))
+      .flatMap(_.set(vulnerableYesNoPage, entity.vulnerableYesNo))
+      .flatMap(answers => extractDateOfBirth(entity.dateOfBirth, answers))
+      .flatMap(answers => extractShareOfIncome(entity.income, answers))
+      .flatMap(answers => extractCountryOfNationality(entity.nationality, answers))
+      .flatMap(answers => extractIdentification(entity.identification, entity.address, answers))
+      .flatMap(answers => extractCountryOfResidence(entity.countryOfResidence, answers))
+      .flatMap(answers => extractAddress(entity.address, answers))
+  }
+
```

Comment:
I would keep this in the `IndividualBeneficiaryExtractor` as it's only specific to that

Reply:
Updated